x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpSettings.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.http;
+
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.core.TimeValue;
+
+import java.util.List;
+
+public class HttpSettings {
+    public static final Setting<ByteSizeValue> MAX_HTTP_RESPONSE_SIZE = Setting.byteSizeSetting(
+        "xpack.inference.http.max_response_size",
+        new ByteSizeValue(10, ByteSizeUnit.MB),   // default
+        ByteSizeValue.ONE, // min
```

Comment:
1 byte essentially means all responses are rejected. User beware! I can't think of a reason to set this to 1, and I think it is an advanced setting so I'm happy to allow the user to do this if they want. Let's make note to document these settings.

Reply:
Yeah good point, this is what watcher had, so I just copied that. Having it as 1 does make some testing easier at the moment but after refactoring from your other comment maybe that wont apply anymore.